test/test_char_array.cpp

```diff
@@ -65,3 +96,63 @@ TEST(test_char_array, resize) {
   // cleanup only 3 fields
   EXPECT_EQ(RCUTILS_RET_OK, rcutils_char_array_fini(&char_array));
 }
+
+TEST_F(ArrayCharTest, vsprintf_fail) {
+  auto failing_allocator = get_failing_allocator();
+  auto ret = rcutils_char_array_init(&char_array, 10, &allocator);
+  ASSERT_EQ(RCUTILS_RET_OK, ret);
+
+  setlocale(LC_ALL, ".932");
```

Comment:
I added a comment, but this was a tricky one needed to make the snprintf function fail. I took the idea from this [stackoverflow post](https://stackoverflow.com/questions/28096802/what-is-the-meaning-of-encoding-error-in-the-vsnprintf-documentation). Not sure if it would be a good idea to include it.

Reply:
Comment is good enough for me.